openmc/region.py

```diff
@@ -229,14 +229,27 @@ def clone(self, memo=None):
         clone : openmc.Region
             The clone of this region
 
-        Raises
-        ------
-        NotImplementedError
-            This method is not implemented for the abstract region class.
+        """
+        pass
+
+    @abstractmethod
+    def translate(self, vector, memo=None):
+        """Translate region in given direction
+
+        Parameters
+        ----------
+        vector : iterable of float
+            Direction in which region should be translated
+        memo : dict or None
```

Comment:
For these memo parameter descriptions, I recommend we do something similar to what we do in clone:
`This parameter is used internally and should not be specified by the user.`